src/ruby/ext/grpc/extconf.rb

```diff
@@ -24,10 +24,18 @@
 
 ENV['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
 
-ENV['AR'] = RbConfig::CONFIG['AR'] + ' rcs'
-ENV['CC'] = RbConfig::CONFIG['CC']
-ENV['CXX'] = RbConfig::CONFIG['CXX']
-ENV['LD'] = ENV['CC']
+if ENV['AR'].nil?
```

Comment:
@apolcyn Thank you for taking a look at this.  I can see the merits for that, however I believe that it's probably the weaker solution for the following reasons:

- it is pretty rare and difficult to accidentally override these variables.  if they are overridden, the user probably meant to do it, and in the unlikely event that they didn't mean to, far more than just grpc will be broken.  it's more than fair to say that messing with the build toolchain puts any issues you might have at the bottom of the priority list.
- `CFLAGS`, `CC` and so-forth have well-known purposes that are standardized across the C/C++ ecosystem.  nonstandard variables would require documentation and the assumption that users will actually find and read said documentation, in the same way that users should be able to expect that `make` with no targets will reasonably build the project, rather than having to read the documentation for some weird invocation.
- it breaks package managers that rely on these standardized variables to control the build environment, such as portage (the original reason for this PR).

Reply:
Thanks for explanation, I can get on board with this.

Reply:
I'm fine with it too.